File: redbus-map/src/main/java/com/redbus/map/ui/MapViewActivity.java
package com.redbus.map.ui;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.redbus.map.R;
import com.redbus.map.batterystatus.RBusBatteryManager;

public class MapViewActivity extends AppCompatActivity {

    private static final String TAG  = "MapViewActivity";

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //RedBusMapProvider.getInstance(this);


    }
}

File: app/src/main/java/com/bus/map/demo/ui/MapViewActivity.kt
package com.bus.map.demo.ui

import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.bus.map.demo.R
import com.bus.map.demo.databinding.ActivityMainBinding
import com.bus.map.demo.module.RedBusMapProvider

class MapViewActivity : BaseActivity() {

    lateinit var binding: ActivityMainBinding
    lateinit var viewModel: MapViewModel
    var mapProviderImpl: RedBusMapProvider = RedBusMapProvider(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        initViewModel()
        binding.viewModel = viewModel
        mapProviderImpl.addMapToView(binding.mapFragmentContainer.id)
        initObservers()

    }

    private fun initViewModel() {
        viewModel = ViewModelProviders.of(this).get(MapViewModel::class.java)
        viewModel.init(this, mapProviderImpl)
    }

    private fun initObservers() {
        viewModel.networkCheckFailed?.observe(this, networkEventObseerver)
        viewModel.errorState?.observe(this, errorStateObserver)
    }

    val errorStateObserver = Observer<String>{message ->
        displayErrorToast(message)
    }
    val networkEventObseerver = Observer<Boolean> { vt ->
        displayNetworkErrorDialog()
    }
}
